lib/OpenQA/Worker/Cache.pm

```diff
@@ -129,6 +129,16 @@ sub download_asset {
             $asset = undef;
         }
     }
+    elsif ($tx->res->code && $tx->res->code =~ /5[0-9]{2}/) {
```

Comment:
You can use the `is_server_error` method for this. http://mojolicious.org/perldoc/Mojo/Message/Response#is_server_error